discovery/gossiper.go

```diff
@@ -107,7 +107,7 @@ type Config struct {
 	// NotifyWhenOnline is a function that allows the gossiper to be
 	// notified when a certain peer comes online, allowing it to
 	// retry sending a peer message.
-	NotifyWhenOnline func(peer *btcec.PublicKey, connectedChan chan<- struct{})
+	NotifyWhenOnline func(peer *btcec.PublicKey, peerChan chan<- lnpeer.Peer)
```

Comment:
Perhaps make a note in the godoc that `peerChan` MUST be buffered

Reply:
Fixed.